src/addressbook.rs

```diff
@@ -0,0 +1,116 @@
+//! # Addressbook utils
+
+use super::core::Address;
+use glob::glob;
+use serde_json;
+use std::fs::File;
+use std::fs::remove_file;
+use std::path::{Path, PathBuf};
+
+/// Addressbook Service
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Addressbook {
+    dir: PathBuf,
+}
+
+/// Addressbook Errors
+#[derive(Debug, Clone)]
+pub enum AddressbookError {
+    /// IO Error
+    IO,
+
+    /// Invalid Address
+    InvalidAddress,
+}
+
+impl Addressbook {
+    /// Initialize new addressbook service for a dir
+    pub fn new(dir: PathBuf) -> Addressbook {
+        Addressbook { dir: dir }
+    }
+
+    fn read_json(path: &Path) -> Result<serde_json::Value, AddressbookError> {
```

Comment:
Our prefer way, for example, ` pub fn search_by_address<P: AsRef<Path>>(addr: &Address, path: P) -> Result<KeyFile, Error>`.